git-open

```diff
@@ -199,6 +199,15 @@ elif [[ "${#pathargs[@]}" -ge 3 && ${pathargs[${#pathargs[@]} - 3]} == 'scm' ]];
   pathargs=(${pathPref[@]} 'projects' ${pathargs[${#pathargs[@]} - 2]} 'repos' "${pathargs[@]:${#pathargs[@]} - 1}")
   IFS='/' urlpath="${pathargs[*]}"
   providerBranchRef="/browse?at=$remote_ref"
+elif [[ "$domain" == *"bitbucket"* ]] && [[ $gitprotocol == 'ssh' ]]; then
```

Comment:
Is this for bitbucket.org or Bitbucket Server (Self hosted). Can we get a test or at the very least an example URL -> what it should be converted to?

Reply:
@paulirish In my case, it is for self-hosted bitbucket. I mentioned the example urls in the issue. Also added a test for the same. Please review it once again.

Reply:
I have the same use case using a self-hosted bitbucket server but `bitbucket` isn't in the domain name so this would still not work for me. I wonder if there's a way to add a flag or update to this PR that uses the same functionality but doesn't assume that `bitbucket` is in the domain.

Reply:
I was wondering if this was the case. Currently the way it looks like we determine if it's a bitbucket server is if it has `scm` in the URL (third to last segment). Is this still valid but only with non-ssh?

Either way the way we may already support this is with the existing `open.` [config params](https://github.com/paulirish/git-open/blob/master/git-open.1.md#gitlab-options).


It's not gitlab specific and people have used it for other providers, can it be used here?

Reply:
@derimagia Yes, current way of determining bitbucket server does not work with ssh. It works with the code I pushed in this PR. 

The documentation is not clear to use config param for bitbucket if it is already supported.

Remote url in git config: ssh://git@bitbucket.scm.xyz.com/aaa/bbb.git
URL to be opened: https://bitbucket.scm.xyz.com/projects/aaa/repos/bbb/browse

I tried these commands, still it did not work
```
git config "open.https://bitbucket.scm.xyz.com.domain" "bitbucket.scm.xyz.com/projects"
git config "open.https://bitbucket.scm.xyz.com.domain" "bitbucket.scm.xyz.com"
git config "open.https://bitbucket.scm.xyz.com.protocol" "https"
```